daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -64,12 +64,18 @@ private[lf] object Compiler {
   case object NoPackageValidation extends PackageValidationMode
   case object FullPackageValidation extends PackageValidationMode
 
+  /** @param enableFullAnfTransformation When true, the SExpr produced by the ANF pass do not contain any
+    *     SEAppOnlyFunIsAtomic applications, only SEAppAtomic ones. This simplifies the code of the speedy machine and
+    *     speeds it up, but also changes the evaluation order or daml programs in a way that is incompatible with the
+    *     daml 2 specification. This incompatibility is observable in the presence of exceptions or non-termination.
+    */
   final case class Config(
       allowedLanguageVersions: VersionRange[LanguageVersion],
       packageValidation: PackageValidationMode,
       profiling: ProfilingMode,
       stacktracing: StackTraceMode,
       enableContractUpgrading: Boolean = false,
+      enableFullAnfTransformation: Boolean = false,
```

Comment:
Following our earlier discussion.... We should promote this new flag as controlling evaluation order, and refrain from even mentioning ANF in it's name.

Being able to do full ANF is the technical benefit of using right-to-left evaluation order, but from the user POV, the semantic change is the changed evaluation order.

And also perhaps we can avoid boolean blindness by taking a similar approach to the other `Mode` controlling flags. Perhaps:
```
      evalOrder: EvalOrderMode = EvalLeftToRight,
```

With:
```
  sealed abstract class EvalOrderMode extends Product with Serializable
  case object EvalLeftToRight extends EvalOrderMode // original order
  case object EvalRightToLeft extends EvalOrderMode // new order: more efficient; less suprising semantics
```

I realise it might be quite painful to change this everywhere, but it would be nice if our internal naming is consistent with the naming at which this flag gets exposed to the user.

Reply:
Done